.github/workflows/pull_request.yml

```diff
@@ -93,18 +102,32 @@ jobs:
             sudo dpkg -i wkhtmltox_0.12.6-1.bionic_amd64.deb
       - name: Write app version
         run: printf "${{ github.event.pull_request.head.sha }}" > .application-version
-      - name: Install pipenv
-        run: pip install pipenv==2023.8.22
-      - name: Install virtual environment
-        run: pipenv install --dev
+      - name: Install poetry
+        uses: snok/install-poetry@v1
+        with:
+          version: 1.7.1
+          virtualenvs-create: true
+          virtualenvs-in-project: false
+          virtualenvs-path: ~/.virtualenvs
+      - uses: actions/setup-python@v4
+        id: cache-poetry
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+          cache: "poetry"
+      - name: Install dotenv plugin
+        run: poetry self add poetry-plugin-dotenv
+      - name: Install Dependencies 
+        if: steps.cache-poetry.outputs.cache-hit != 'true'
```

Comment:
This shouldn't be needed. See https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#caching-packages

Check if poetry action is compatible with this, if not we can install via `pipx`

Reply:
I think I have updated it as you suggested. I was a bit confused so could you check if it was the change you wanted 👍